lib-injection/sitecustomize.py

```diff
@@ -159,15 +170,21 @@ def package_is_compatible(package_name, package_version):
     return installed_version.version >= supported_version_spec.version
 
 
+def executable_is_compatible():
```

Comment:
Typically, argv elements won't get expanded to their full paths.  It may be better to
1. truncate to the basename on both sides (the deny list and the argv element we're looking at)
2. compare the basenames
3. Optionally--and only if the path is unqualified (for example, not `./foo.py`)--if there's a match, expand `$PATH` and check whether the basename identifies a file in each directory of `$PATH`

This is imperfect because it doesn't use the exact same resolution mechanism as the underlying shell or executor.  I suppose an equivalent-or-better solution may be to `popen()` to `sh` and run `command -v` (which is part of POSIX and thus should be available as a builtin in the implementation of `sh`) to verify that the basename is part of the path.

I guess you should just interpret this comment to suggest that maybe the representation of the deny list as absolute paths may be working against our goal here.

Reply:
Thanks. I updated the changeset to check both the given `argv[0]` and the basename'd version.